docs/codegen/overview.md

```diff
@@ -4,32 +4,70 @@ Code generation is the main way you can glue [Terramate data](../sharing-data.md
 to other tools, like Terraform.
 
 Different code generation strategies will be provided in the future to support
-each integration scenario in the best way possible. Currently, we support:
+each integration scenario in the best way possible. 
 
-* [HCL generation](./generate-hcl.md)
-* [File generation](./generate-file.md)
+Currently, we support:
 
-Some features are available to all code generation strategies, like:
+* [HCL generation](./generate-hcl.md) with stack [context](#generation-context).
+* [File generation](./generate-file.md) with `root` and `stack` [context](#generation-context).
+
+# Generation Context
+
+The Terramate code generation supports two execution context:
+
+- stack: generates code relative to the stack where it's defined.
+- root: generates code outside of stacks.
+
+The `stack` context gives access to all code generation features, like:
 
 * [Globals](../sharing-data.md#globals)
-* [Metadata](../sharing-data.md#metadata)
+* [All Metadata](../sharing-data.md#metadata)
 * [Functions](../functions.md)
 * [Lets](#lets)
 * [Assertions](#assertions)
 
+But the `root` context gives access to:
+
+* [Project Metadata](../sharing-data.md#project-metadata)
+* [Functions](../functions.md)
+* [Lets](#lets)
+
+If not specified otherwise, the default generation context is `stack`.
+For the `generate_hcl` there's no way to change the `context` and it's always
+of type `stack` but the `generate_file` has a `context` attribute which you can explicit change it to `root`. Example:
+
+```hcl
+generate_file "/file.txt" {
+    context = root
+    content = "something"
+}
+```
+
 # Labels
 
 All code generation blocks use labels to identify the block and define where
-the generated code will be saved. The labels must follow these constraints:
+the generated code will be saved but they have different constraints depending
+on the [generation context](#generation-context).
 
-* It is a relative path in the form `dir/filename` or just `filename`
+For `stack` context, the labels must follow the constraints below:
+
+* It is a relative path in the form `<dir>/<filename>` or just `<filename>`
 * It is always defined with `/` independent on the OS you are working on
 * It does not contain `../` (code can only be generated inside the stack)
 * It does not start with `./`
 * It is not a symbolic link
 * It is not a stack
 * It is unique on the whole hierarchy of a stack
 
+For `root` context, the constraints are:
+
+* It is an absolute path in the form `/<dir>/<filename>` or just `/<filename>`.
+* It is always defined with `/` independent on the OS you are working on
+* It does not contain `../` (code can only be generated inside the project root)
+* It is not a symbolic link
+* It is not a stack
+* It is unique on the whole hierarchy for all blocks with condition=true.
```

Comment:
Maybe add "for all blocks with condition=true" also on the stack context docs (sorry I missed that).